.github/workflows/lint-test.yml

```diff
@@ -33,144 +38,228 @@ jobs:
     # https://pdm-project.org/en/latest/usage/lockfile/
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v4
-
-    # Instead of setup-env, we call the cache-pdm action here directly.
-    # This avoids having to install PDM, only to find out the cache is already up to date sometimes.
-    - name: Configure cache
-      uses: ./.github/actions/cache-pdm
-      with:
-        env-already-initialized: false
-
+      - uses: actions/checkout@v4
+
+      - name: Set up uv to lock dependencies
+        id: setup-python
+        uses: astral-sh/setup-uv@v6
+        with:
+          version: ${{ env.USE_UV_VERSION }}
+          enable-cache: true
+
+      - name: Restore uv lock
+        uses: actions/cache/restore@v4
+        id: cache-uv-lock
+        with:
+          key: uv-lock-resolved-${{ env.UV_RESOLUTION }}
+          restore-keys: |
+            uv-lock-resolved-${{ env.UV_RESOLUTION }}
+          path: uv.lock
+          enableCrossOsArchive: true
+
+      - name: Check if uv.lock changed
+        id: uv-lock-changed
+        continue-on-error: true
+        if: steps.cache-uv-lock.outputs.cache-hit == 'true' || steps.cache-uv-lock.outputs.cache-hit == 'false'
+        run: |
+          uv lock --check
+
+      - name: Lock dependencies
+        if: steps.uv-lock-changed.outcome == 'failure' || steps.cache-uv-lock.outputs.cache-hit != 'true' && steps.cache-uv-lock.outputs.cache-hit != 'false'
+        id: lock-uv-deps
+        env:
+          UV_LOCKED: 0
+        run: uv lock --refresh
+
+      - uses: actions/cache/save@v4
+        if: steps.lock-uv-deps.outcome == 'success'  && steps.uv-lock-changed.outcome == 'failure'
+        name: upload-uv-lock
+        with:
+          key: uv-lock-resolved-${{ env.UV_RESOLUTION}}-${{hashFiles('uv.lock')}}
+          path: uv.lock
+          enableCrossOsArchive: true
+
+  # used for pyright and tests
+  determine-python-versions:
+    runs-on: ubuntu-latest
+    outputs:
+      test-python-versions: ${{ steps.set-matrix.outputs.test-python-versions }}
+      libcst-python-version: ${{ steps.set-matrix.outputs.libcst-python-version }}
+      misc-checks-python-version: ${{ steps.set-matrix.outputs.misc-checks-python-version }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install uv
+        uses: astral-sh/setup-uv@v6
+        with:
+          version: ${{ env.USE_UV_VERSION }}
+          enable-cache: true
+          python-version: 3.8
+
+      - name: Determine Python versions to test and lint against
+        id: set-matrix
+        run: |
+          echo test-python-versions=$(uv run noxfile.py -s test --list --json | jq -c '[.[].python]') | tee --append $GITHUB_OUTPUT
+          echo libcst-python-version=$(uv run noxfile.py -s codemod --list --json | jq -rc '[.[].python].[0]') | tee --append $GITHUB_OUTPUT
```

Comment:
This only works as long as the job uses python < 3.11, nox returns `"python": null` for me running on 3.12